process/headerCheck/headerSignatureVerify_test.go

```diff
@@ -0,0 +1,342 @@
+	t.Parallel()
+
+	args := createHeaderSigVerifierArgs()
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{}
+
+	err := hdrSigVerifier.VerifyRandSeed(header)
+	require.Equal(t, sharding.ErrNilRandomness, err)
+}
+
+func TestHeaderSigVerifier_VerifyRandSeedOk(t *testing.T) {
+	t.Parallel()
+
+	args := createHeaderSigVerifierArgs()
+	wasCalled := false
+
+	args.KeyGen = &mock.SingleSignKeyGenMock{
+		PublicKeyFromByteArrayCalled: func(b []byte) (key crypto.PublicKey, err error) {
+			return &mock.SingleSignPublicKey{}, nil
+		},
+	}
+	args.SingleSigVerifier = &mock.SignerMock{
+		VerifyStub: func(public crypto.PublicKey, msg []byte, sig []byte) error {
+			wasCalled = true
+			return nil
+		},
+	}
+
+	pkAddr := []byte("aaa00000000000000000000000000000")
+	nodesCoordinator := &mock.NodesCoordinatorMock{
+		ComputeValidatorsGroupCalled: func(randomness []byte, round uint64, shardId uint32) (validators []sharding.Validator, err error) {
+			v, _ := sharding.NewValidator(big.NewInt(0), 1, pkAddr, pkAddr)
+			return []sharding.Validator{v}, nil
+		},
+	}
+	args.NodesCoordinator = nodesCoordinator
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{}
+
+	err := hdrSigVerifier.VerifyRandSeed(header)
+	require.Nil(t, err)
+	require.True(t, wasCalled)
+}
+
+func TestHeaderSigVerifier_VerifyRandSeedShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := createHeaderSigVerifierArgs()
+	wasCalled := false
+	localError := errors.New("err")
+
+	args.KeyGen = &mock.SingleSignKeyGenMock{
+		PublicKeyFromByteArrayCalled: func(b []byte) (key crypto.PublicKey, err error) {
+			return &mock.SingleSignPublicKey{}, nil
+		},
+	}
+	args.SingleSigVerifier = &mock.SignerMock{
+		VerifyStub: func(public crypto.PublicKey, msg []byte, sig []byte) error {
+			wasCalled = true
+			return localError
+		},
+	}
+
+	pkAddr := []byte("aaa00000000000000000000000000000")
+	nodesCoordinator := &mock.NodesCoordinatorMock{
+		ComputeValidatorsGroupCalled: func(randomness []byte, round uint64, shardId uint32) (validators []sharding.Validator, err error) {
+			v, _ := sharding.NewValidator(big.NewInt(0), 1, pkAddr, pkAddr)
+			return []sharding.Validator{v}, nil
+		},
+	}
+	args.NodesCoordinator = nodesCoordinator
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{}
+
+	err := hdrSigVerifier.VerifyRandSeed(header)
+	require.Equal(t, localError, err)
+	require.True(t, wasCalled)
+}
+
+func TestHeaderSigVerifier_VerifyRandSeedAndLeaderSignatureNilRandomnessShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := createHeaderSigVerifierArgs()
+	hdrSigVerifier, _ := NewHeaderSigVerifier(args)
+	header := &dataBlock.Header{}
+
+	err := hdrSigVerifier.VerifyRandSeedAndLeaderSignature(header)
+	require.Equal(t, sharding.ErrNilRandomness, err)
+}
+
+func TestHeaderSigVerifier_VerifyRandSeedAndLeaderSignatureVerifyShouldErr(t *testing.T) {
```

Comment:
rename to `TestHeaderSigVerifier_VerifyRandSeedAndLeaderSignatureVerifyShouldErrWhenValidationFails`

Reply:
done.